.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,160 @@
+### Contributing to ZFS on Linux
+
+<p align="center"><img src="http://zfsonlinux.org/images/zfs-linux.png"/></p>
+
+*First of all, thank you for taking the time to contribute!*
+
+The following is a set of guidelines for contributing to ZFS on Linux. 
+Using them you can help us to make a better work.
+
+#### Table Of Contents
+
+[What should I know before I get started?](#what-should-i-know-before-i-get-started)
+
+  * [Get ZFS](#get-zfs)
+  * [Debugging ZFS](#debugging-zfs)
+  * [Where can I ask for help?](#where-can-I-ask-for-help)
+  
+[How Can I Contribute?](#how-can-i-contribute)
+
+  * [Reporting Bugs](#reporting-bugs)
+  * [Suggesting Enhancements](#suggesting-enhancements)
+  * [Pull Requests](#pull-requests)
+  * [Testing](#testing)
+  
+[Styleguides](#styleguides)
+ 
+  * [Coding conventions](#coding-conventions)
+
+Helpful resources
+
+  * [ZFS on Linux wiki](https://github.com/zfsonlinux/zfs/wiki)
+  * [OpenZFS Documentation](http://open-zfs.org/wiki/Developer_resources)
+
+
+## What should I know before I get started?
+
+### Get ZFS
+
+You can build fresh zfs packages using 
+[this instruction](https://github.com/zfsonlinux/zfs/wiki/Building-ZFS), 
+or use stable packages from 
+[your distribution repository](https://github.com/zfsonlinux/zfs/wiki/Getting-Started).
+
+### Debugging ZFS
+
+`zdb` utility is the main tool to debug ZFS. 
+
+Additionally, see 
+[build instruction](https://github.com/zfsonlinux/zfs/wiki/Building-ZFS)
+, there is `--enable-debug` flag, which can be set to enable all ASSERTs and additional correctness tests.
+
+### Where can I ask for help?
+The best place to ask is our [mailing list](https://github.com/zfsonlinux/zfs/wiki/Mailing-Lists).
+
+Please *note*, our issue tracker is **only for bugs**, if you have thoughts about any 
+enhancements - first tell about them via 
+[mailing list](https://github.com/zfsonlinux/zfs/wiki/Mailing-Lists).
+
+## How Can I Contribute?
+
+### Reporting Bugs
+
+You can open a new issue and search existing issues using the public
```

Comment:
I would encourage people to search first. This will help cut down on duplicate issues. Maybe something like `If you run into an issue, please search our issue tracker to ensure the issue hasn't been reported before. If you are seeing a new issue, please create a new issue.` Thoughts?